### PR TITLE
Hash dataplane volume names when bigger than DNS label

### PR DESCRIPTION
**Pull request by fao89**
closes [OSPRH-8801](https://issues.redhat.com//browse/OSPRH-8801)

### CODE REVIEW

File: pkg/dataplane/deployment.go [hunk 1]
```diff
@@ -313,7 +314,8 @@ func (d *Deployer) addCertMounts(
 				}
 				volumeName := GetServiceCertsSecretName(d.NodeSet, service.Name, certKey, 0)
 				if len(volumeName) > apimachineryvalidation.DNS1123LabelMaxLength {
-					volumeName = volumeName[:apimachineryvalidation.DNS1123LabelMaxLength]
+					hash := sha256.Sum224([]byte(volumeName))
+					volumeName = string(hash[:])
```
Comment (olliewalsh) [inline pkg/dataplane/deployment.go:318]: doesn't produce a hex string, `hex.EncodeToString(hash[:])` from enconding/hex works

Comment (olliewalsh) [reply pkg/dataplane/deployment.go:318]: since the hex string length is 56 maybe can add a short prefix e.g "cert", "cacert", "cm", "sec"

**Review by olliewalsh — DISMISSED** [review]

File: pkg/dataplane/deployment.go [hunk 2]
```diff
@@ -412,8 +416,8 @@ func (d *Deployer) addServiceExtraMounts(
 		for idx, key := range keys {
 			volumeName := fmt.Sprintf("%s-%s", cm.Name, strconv.Itoa(idx))
 			if len(volumeName) > apimachineryvalidation.DNS1123LabelMaxLength {
-				limit := apimachineryvalidation.DNS1123LabelMaxLength - len(strconv.Itoa(idx))
-				volumeName = volumeName[:limit] + strconv.Itoa(idx)
+				hash := sha256.Sum224([]byte(volumeName))
+				volumeName = "cm" + hex.EncodeToString(hash[:]) + strconv.Itoa(idx)
```
Comment (olliewalsh) [inline pkg/dataplane/deployment.go:420]: any reason to keep the appended index?

Comment (olliewalsh) [reply pkg/dataplane/deployment.go:420]: if it's intentional I would add a `-` or it will look like it's part of the hash

**Review by bshephar — APPROVED** [review]

**Review by olliewalsh — APPROVED** [review]

**Pull request merged**